core/src/main/scala/kafka/utils/Logging.scala

```diff
@@ -17,106 +17,74 @@
 
 package kafka.utils
 
-import org.apache.log4j.Logger
+import com.typesafe.scalalogging.{LazyLogging, Logger}
+import org.slf4j.{Marker, MarkerFactory}
 
-trait Logging {
-  val loggerName = this.getClass.getName
-  lazy val logger = Logger.getLogger(loggerName)
 
-  protected var logIdent: String = null
+object Log4jControllerRegistration {
+  private val logger = Logger(this.getClass.getName)
 
-  // Force initialization to register Log4jControllerMBean
-  private val log4jController = Log4jController
+  try {
+    val log4jController = Class.forName("kafka.utils.Log4jController").asInstanceOf[Class[Object]]
+    val instance = log4jController.getDeclaredConstructor().newInstance()
```

Comment:
Nit: unnecessary `()`

Reply:
As I see I need this as in Java 8 both `getDeclaredConstructor` and `newInstance` are variadic methods, so they actually accept a list of object, therefore Scala won't allow me to use them without `()`.